geopandas/geodataframe.py

```diff
@@ -214,10 +214,33 @@ def __setattr__(self, attr, val):
 
     def _get_geometry(self):
         if self._geometry_column_name not in self:
-            raise AttributeError(
-                "No geometry data set yet (expected in"
-                " column '%s'.)" % self._geometry_column_name
-            )
+            if self._geometry_column_name is None:
+                msg = (
+                    "You are calling a geospatial method on the GeoDataFrame, "
+                    "but the default active geometry column to use has not been set. "
+                )
+            else:
+                msg = (
+                    "You are calling a geospatial method on the GeoDataFrame, "
+                    f"but the active geometry column ({self._geometry_column_name}) "
+                    "is no longer present. "
+                )
+            geo_cols = list(self.columns[self.dtypes == "geometry"])
+            if len(geo_cols) > 0:
+                msg += (
+                    f"\nThere are columns with geometry data type ({geo_cols}), and "
+                    "you can either set this as the active geometry with "
+                    'df.set_geometry("name") or access this column as a '
```

Comment:
```suggestion
                    "you can either set one as the active geometry with "
                    'df.set_geometry("name") or access the column as a '
```
"columns ... this" doesn't work. Same below, we may talk about multiple columns.